src/program-structure-and-compilation.rst

```diff
@@ -114,21 +114,30 @@ A :t:`crate` is a unit of compilation and linking that contains a tree of
 nested :t:`[module]s`.
 
 :dp:`fls_9ub6ks8qrang`
-A :t:`binary crate` is a :t:`crate` that contains a :t:`main function`. A tool
-can compile a :t:`binary crate` to an executable.
+A :t:`binary crate` is a :t:`crate` that has a :t:`function` in scope of its
+:t:`crate root module` under the :t:`name` ``main`` with a :t:`main function
+signature`. A tool can compile a :t:`binary crate` to an executable.
+
+:dp:`fls_jQqXxPyND1ds`
+The :t:`function` in scope of a :t:`binary crate`'s :t:`crate root module` under
+the :t:`name` ``main`` with a :t:`main function signature` is the :t:`binary
+crate`'s :t:`program entry point`.
 
 :dp:`fls_d9nn4yuiw1ja`
-A :t:`library crate` is either a :t:`crate` without a :t:`main function` or a
-:t:`crate` subject to :t:`attribute` :c:`no_main`. A tool is free to compile a
-:t:`library crate` to a shared library.
+A :t:`library crate` is either a :t:`crate` that does not have a :t:`function`
+in scope of its :t:`crate root module` under the :t:`name` ``main`` with a
+:t:`main function signature` or a :t:`crate` subject to :t:`attribute`
+:c:`no_main`. A tool is free to compile a :t:`library crate` to a shared
+library.
```

Comment:
I am quite confused by this: whether a crate is a binary crate or a library crate is just determined by the `--crate-type` flag or the `#![crate_type]` attribute. You can also have executables with `#![no_main]` (if you define your own entry point the linker can find).